packages/@react-aria/menu/src/useMenuItem.ts

```diff
@@ -80,7 +84,9 @@ export interface AriaMenuItemProps {
    * Handler that is called when the user activates the item.
    * @deprecated - pass to the menu instead.
    */
-  onAction?: (key: Key) => void
+  onAction?: (key: Key) => void,
+  isMenuDialogTrigger?: boolean,
```

Comment:
I'm unsure how to accomplish this. I could introduce a new 'type' to return from the yield for MenuDialogTrigger's getCollectionNode, however, I don't think it'll come through CollectionBuilder to TreeCollection where we iterate over all the nodes.

I can't set hasChildItems on the yield of MenuDialogTrigger either. Since it doesn't end up on the Item node and itself doesn't end up in node list due to it being a wrapper.

I can however use hasChildItems directly on Item, but that requires users to do that as well. I've gone that way for now, even though it feels odd because the Item doesn't actually have child items, it has a sibling.

Reply:
I guess you could cloneElement within the getCollectionNode and add `hasChildItems` to the `Item`. Alternatively you could add it to the yielded object, and make some change to CollectionBuilder to propagate it to the child but not sure its worth it.

Reply:
Right, I thought about putting it on via cloneElement, but got a little hung up on supporting wrapping elements around Item. Though right now we don't support that anyways and in the new collections we can do this more easily and wouldn't need to expose it to users there either. Maybe cloneElement is the best for this for now.